### PR TITLE
Json api: Add new plugin/new api endpoint

### PR DESCRIPTION
**Pull request by enejb**
This api endpoint allows us to install a plugin via by uploading a zip file.

For a description how to test this PR 
Follow the instructions here D3262-code

### CODE REVIEW

File: json-endpoints/jetpack/class.jetpack-json-api-plugins-new-endpoint.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			if ( empty( $plugin ) ) {
+				return new WP_Error( 'plugin_already_installed' );
+			}
+
+			$this->plugins = $plugin;
+			$this->log[ $plugin[0] ] = $upgrader->skin->get_upgrade_messages();
+			return true;
+		}
+
+		return new WP_Error( 'No Plugin installed' );
```
Comment (seear) [inline json-endpoints/jetpack/class.jetpack-json-api-plugins-new-endpoint.php]: Dodgy capitalization on this message :)

File: json-endpoints/jetpack/json-api-jetpack-endpoints.php [hunk 2]
```diff
@@ -240,6 +240,28 @@
 ) );
 
 require_once( $json_jetpack_endpoints_dir . 'class.jetpack-json-api-plugins-modify-endpoint.php' );
+require_once( $json_jetpack_endpoints_dir . 'class.jetpack-json-api-plugins-new-endpoint.php' );
+// POST /sites/%s/plugins/new
+new Jetpack_JSON_API_Plugins_New_Endpoint( array(
+	'description'     => 'Install a plugin to a Jetpack site by uploading a zip file',
+	'group'           => '__do_not_document',
+	'stat'            => 'plugins:new',
```
Comment (lezama) [inline json-endpoints/jetpack/json-api-jetpack-endpoints.php]: 👍

**Comment by enejb** [comment]
https://github.com/Automattic/jetpack/issues/5466

**Comment by lezama** [comment]
🚢

**Comment by dereksmart** [comment]
Needs a rebase since #5537 was merged 😬

**Comment by seear** [comment]
> Needs a rebase since #5537 was merged 

Rebasing now...

**Pull request merged**